src/index.ts

```diff
@@ -149,11 +191,30 @@ export { extractPublicKey } from './utils.js'
  */
 const signLegacyV1 = async (privateKey: PrivateKey, value: Uint8Array, validityType: IpnsEntry.ValidityType, validity: Uint8Array): Promise<Uint8Array> => {
   try {
-    const dataForSignature = ipnsEntryDataForV1Sig(value, validityType, validity)
+    const dataForSignature = ipnsRecordDataForV1Sig(value, validityType, validity)
 
     return await privateKey.sign(dataForSignature)
   } catch (error: any) {
     log.error('record signature creation failed', error)
     throw errCode(new Error('record signature creation failed'), ERRORS.ERR_SIGNATURE_CREATION)
   }
 }
+
+/**
+ * Normalizes the given record value. It ensures it is a string starting with '/'.
+ * If the given value is a cid, the returned path will be '/ipfs/{cid}'.
+ */
+const normalizeValue = (value: string | Uint8Array): string => {
+  const str = typeof value === 'string' ? value : uint8ArrayToString(value)
+
+  if (str.startsWith('/')) {
+    return str
+  }
+
+  try {
+    const cid = CID.parse(str)
+    return '/ipfs/' + cid.toV1().toString()
+  } catch (_) {
```

Comment:
If ignoring the error the following can be used:

```suggestion
  } catch {
```

Though I think we should at least log it here.

Better yet, accept a `CID` argument so we re-use the validation from that class, then this try/catch can be removed.